### ISSUE TITLE
Java inheritance is missing

### ISSUE THREAD

**Issue by stuartmorgan-g** [dart-lang/native#1653]
I'm running `jnigen` on `public interface ExoPlayer extends Player {...}` The resulting Dart object is `class ExoPlayer extends _$jni.JObject {...}`, with none of the methods from `Player`.

I thought the problem was that I wasn't generating `Player`, so I added that to my config (although I would have filed that as an issue if that were it; this should Just Work IMO), but it didn't help; I have the `Player` class now too, but `ExoPlayer` is still just `class ExoPlayer extends _$jni.JObject` and didn't get any new methods.

**Comment by HosseinYousefi** [comment]
Will be solved by: https://github.com/dart-lang/native/issues/577. I will prioritize this.

**Comment by stuartmorgan** [comment]
> This technically shouldn't be, will investigate.

Yes, I saw class inheritance in your prototype, so I was confused by it not being in mine. I wasn't sure if I was doing something non-obviously wrong, or it was something that was implemented on a branch.

**Comment by stuartmorgan** [comment]
FYI I'm not blocked on this; I just made a helper to Java-cast to the superclass. This is definitely an important use case to have working, but no need to change any short-term work planning on my account.

**Comment by HosseinYousefi** [comment]
> Yes, I saw class inheritance in your prototype, so I was confused by it not being in mine. I wasn't sure if I was doing something non-obviously wrong, or it was something that was implemented on a branch.

I think I know why this happened. Probably `extends` for interfaces is equivalent `implements` and we only use `extends` for inheritance.

**Comment by HosseinYousefi** [comment]
> I just made a helper to Java-cast to the superclass.

Possible to do `exoPlayer.as(Player.type)` in Dart to cast as well.

**Comment by stuartmorgan** [comment]
Yes, that's what I meant by Java-cast, sorry for the confusing terminology. The helper I was referring to is Dart helper because I need to do this constantly in the code because, as most of the methods are on the superclass.

**Comment by liamappelbe** [comment]
This was fixed by https://github.com/dart-lang/native/pull/3201. Classes and interfaces now implement all their interfaces.
